examples/zero_inflated_poisson.py

```diff
@@ -0,0 +1,136 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Zero-Inflated Poisson regression model
+================================================
+
+We evaluate Zero-Inflated Poisson regression model with MLE and MCMC in terms of their predictive
+performances of the number of fish caught.
+"""
+
+import argparse
+import os
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.metrics import mean_squared_error
+import statsmodels.api as sm
+
+from jax import random
+import jax.numpy as jnp
+import jax.scipy as jsp
+
+import numpyro
+import numpyro.distributions as dist
+from numpyro.infer import MCMC, NUTS
+from numpyro.infer.util import Predictive
+
+matplotlib.use("Agg")  # noqa: E402
+
+
+def model(X, Y):
+    D_X = X.shape[1]
+    b1 = numpyro.sample("b1", dist.Normal(0.0, 1.0).expand([D_X]).to_event(1))
+    b2 = numpyro.sample("b2", dist.Normal(0.0, 1.0).expand([D_X]).to_event(1))
+
+    q = jsp.special.expit(jnp.dot(X, b1[:, None])).reshape(-1)
+    lam = jnp.exp(jnp.dot(X, b2[:, None]).reshape(-1))
+
+    with numpyro.plate("obs", X.shape[0]):
+        numpyro.sample("Y", dist.ZeroInflatedPoisson(gate=q, rate=lam), obs=Y)
+
+
+def run_inference(model, args, X, Y):
+    kernel = NUTS(model)
+    mcmc = MCMC(
+        kernel,
+        num_warmup=args.num_warmup,
+        num_samples=args.num_samples,
+        num_chains=args.num_chains,
+        progress_bar=False if "NUMPYRO_SPHINXBUILD" in os.environ else True,
+    )
+    mcmc.run(random.PRNGKey(1), X, Y)
+    mcmc.print_summary()
+    return mcmc.get_samples()
+
+
+def fit_statsmodel_zip(args, X, Y):
+    zip_model = sm.ZeroInflatedPoisson(endog=Y, exog=X, exog_infl=X, inflation="logit")
```

Comment:
i think it would be much nicer if instead of using `sm` you used a `AutoDelta` guide and did maximum likelihood estimation within numpyro